scaleway/resource_function_namespace.go

```diff
@@ -0,0 +1,160 @@
+package scaleway
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	function "github.com/scaleway/scaleway-sdk-go/api/function/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayFunctionNamespace() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayFunctionNamespaceCreate,
+		ReadContext:   resourceScalewayFunctionNamespaceRead,
+		UpdateContext: resourceScalewayFunctionNamespaceUpdate,
+		DeleteContext: resourceScalewayFunctionNamespaceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultFunctionNamespaceTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				ForceNew:    true,
+				Optional:    true,
+				Description: "The name of the function namespace",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the function namespace",
+			},
+			"environment_variables": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "The environment variables of the function namespace",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"registry_endpoint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint reachable by docker",
+			},
+			"registry_namespace_id": {
```

Comment:
Should we consider the functions and triggers in this PR? 🤔  and maybe change the resource in only resource_function. (open question)

Reply:
This feature will be handled in the function resource.